pkg/repository/base.go

```diff
@@ -143,6 +143,7 @@ type ProviderRepository interface {
 	SetMachineLock(providerName, poolName, machineId string) error
 	RemoveMachineLock(providerName, poolName, machineId string) error
 	GetGPUAvailability(pools map[string]types.WorkerPoolConfig) (map[string]bool, error)
+	GetGPUCounts(pools map[string]types.WorkerPoolConfig) (map[string]int, error)
```

Comment:
looking at the function above this, do we really need a separate repository function for this?

can't we use the results from GetGPUAvailability (maybe with something else added), and use that in the new endpoint?

Reply:
`GetGPUAvailability` is just returning a map of GPU to boolean values and does not really count. If we just want to say there is no availability at all that could work, but it doesn't give us much flexibility to warn about low availability.